### DISCUSSION TITLE
WiFi wrong password

### DISCUSSION THREAD

**Discussion by maxklimes** [espressif/arduino-esp32#6948] [Q&A]
Hi there everyone,

I'm working on a couple of projects that all include WiFi in one way or another, and I've noticed that there is no way to determine if the user has entered a wrong password when trying to connect to a WiFi network. There seems to be the wifi_station_get_connect_status() function on ESP8266 that can return STATION_WRONG_PASSWORD. However, I've come to discover that this function isn't present on ESP32 and it's functionality should all be in the WiFi.status()function instead. WiFi.status()does not differentiate between a failed connection and a wrong password though. I've tried Googling all around, but can't for the life of me figure out how to determine if a password is wrong. Has anyone else encountered this issue? Maybe a separate library that retains this functionality?

If it's of any help, I'm connecting in STA mode via the ESP.waitForConnectResult() function. Thanks in advance!

**Comment by kemalyilmaz77** [comment]
did you find a solution i have this problem too

**Comment by ziadelbouriny** [comment]
Check [this](https://github.com/ziadelbouriny/WiFi-Repeater/commit/6f6185d681a8689f51884268a21344c26a584fb6)
